src/core/Effect.cpp

```diff
@@ -181,26 +169,56 @@ Effect * Effect::instantiate( const QString& pluginName,
 
 
 
-void Effect::checkGate(double outSum)
+void Effect::handleAutoQuit(std::span<const SampleFrame> output)
 {
-	if( m_autoQuitDisabled )
+	if (m_autoQuitDisabled)
 	{
 		return;
 	}
 
-	// Check whether we need to continue processing input.  Restart the
+	/*
+	 * In the past, the RMS was calculated then compared with a threshold of 10^(-10).
+	 * Now we use a different algorithm to determine whether a buffer is non-quiet, so
+	 * a new threshold is needed for the best compatibility. The following is how it's derived.
+	 *
+	 * Old method:
+	 * RMS = average (L^2 + R^2) across stereo buffer.
+	 * RMS threshold = 10^(-10)
+	 *
+	 * So for a single channel, it would be:
+	 * RMS/2 = average M^2 across single channel buffer.
+	 * RMS/2 threshold = 5^(-11)
+	 *
+	 * The new algorithm for determining whether a buffer is non-silent compares M with the threshold,
+	 * not M^2, so the square root of M^2's threshold should give us the most compatible threshold for
+	 * the new algorithm:
+	 *
+	 * (RMS/2)^0.5 = (5^(-11))^0.5 = 0.0001431 (approx.)
+	 *
+	 * In practice though, the exact value shouldn't really matter so long as it's sufficiently small.
+	 */
+	static constexpr auto threshold = 0.0001431f;
+
+	// Check whether we need to continue processing input. Restart the
 	// counter if the threshold has been exceeded.
-	if (outSum - gate() <= F_EPSILON)
+
+	for (const SampleFrame& frame : output)
 	{
-		incrementBufferCount();
-		if( bufferCount() > timeout() )
+		const auto abs = frame.abs();
+		if (abs.left() >= threshold || abs.right() >= threshold)
 		{
-			stopRunning();
+			// The output buffer is not quiet
 			resetBufferCount();
+			return;
 		}
 	}
-	else
+
+	// The output buffer is quiet, so check if auto-quit should be activated yet
+	incrementBufferCount();
```

Comment:
I dont know if this function is named appropriately, but I guess thats not relevant here.